docker/client.py

```diff
@@ -239,7 +241,7 @@ def _multiplexed_response_stream_helper(self, response):
         # Disable timeout on the underlying socket to prevent
         # Read timed out(s) for long running processes
         socket = self._get_raw_response_socket(response)
-        if six.PY3:
+        if six.PY3 and hasattr(socket, "_sock"):
```

Comment:
Alternatively, what we could do is settimeout(None) on both the socket and socket._sock.